examples/stress_tests/time_smoothness.rs

```diff
@@ -0,0 +1,355 @@
+//! Renders a scrolling background behind an animated sprite at a high zoom
+//! level, to test consistency and smoothness of performance.
+//!
+//! To measure performance realistically, be sure to run this in release mode.
+//! `cargo run --example time_smoothness --release`
+//!
+//! By default, this example scrolls the background at 120 pixels per second,
+//! and always moves in whole-pixel increments (since limiting movement to whole
+//! pixels at high zoom seems to make it easier to perceive any problems with
+//! frame consistency). There are several keyboard controls for changing the
+//! example's behavior at runtime:
+//!
+//! - P: Cycle the `PresentMode` between `Fifo`, `Mailbox`, and `Immediate`.
+//! - W: Cycle the `WindowMode` between `Windowed`, `BorderlessFullscreen`, and
+//!   `Fullscreen`.
+//! - T: Cycle the delta time between normal (measured from the `Time` resource)
+//!   and fixed-interval. This can be useful when trying to distinguish slow
+//!   frametimes from inaccurate time measurement. Fixed-interval delta time is
+//!   hardcoded to 1/60th of a second per frame, which will look highly wacky
+//!   unless you're using Fifo mode on a 60hz display.
+//! - M: Cycle the scrolling motion style between whole-pixel and sub-pixel
+//!   Transform increments.
+//!
+//! A number of factors contribute to scrolling without perceptible
+//! hiccups/stutters/jank, including the accuracy of the delta time measurement,
+//! the ability to consistently present frames to the GPU at the expected pace,
+//! etc. This example doesn't isolate all of those factors, but it can help
+//! identify when a problem exists and provide a starting point for further
+//! investigation.
+use bevy::{
+    diagnostic::{Diagnostic, DiagnosticId, Diagnostics},
+    prelude::*,
+    render::texture::ImageSettings,
+    window::{PresentMode, WindowMode},
+};
+
+// BG_WIDTH is smaller than the image's actual pixel width, because we want the
+// empty space on the background tiles to overlap a bit. That way there's always
+// a "landmark" on screen at the default window size.
+const BG_WIDTH: f32 = 755.0;
+const BG_HEIGHT: f32 = 363.0;
+const BG_TILES: usize = 3;
+const BG_SPEED: f32 = 120.0;
```

Comment:
Maybe add a recommendation here to change this number for different monitors. This is always going to hit the "big jump" warning on a 30Hz monitor and will probably always look jumpy on a 144Hz monitor.